[Bug]: Llama 3.2 90b crash

### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
Collecting environment information...
PyTorch version: 2.5.0+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.12.7 (main, Oct  1 2024, 08:52:12) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.15.0-112-generic-x86_64-with-glibc2.35
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 57 bits virtual
Byte Order:                         Little Endian
CPU(s):                             224
On-line CPU(s) list:                0-223
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Xeon(R) Platinum 8480+
CPU family:                         6
Model:                              143
Thread(s) per core:                 2
Core(s) per socket:                 56
Socket(s):                          2
Stepping:                           8
BogoMIPS:                           4000.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
L1d cache:                          5.3 MiB (112 instances)
L1i cache:                          3.5 MiB (112 instances)
L2 cache:                           224 MiB (112 instances)
L3 cache:                           210 MiB (2 instances)
NUMA node(s):                       2
NUMA node0 CPU(s):                  0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94,96,98,100,102,104,106,108,110,112,114,116,118,120,122,124,126,128,130,132,134,136,138,140,142,144,146,148,150,152,154,156,158,160,162,164,166,168,170,172,174,176,178,180,182,184,186,188,190,192,194,196,198,200,202,204,206,208,210,212,214,216,218,220,222
NUMA node1 CPU(s):                  1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105,107,109,111,113,115,117,119,121,123,125,127,129,131,133,135,137,139,141,143,145,147,149,151,153,155,157,159,161,163,165,167,169,171,173,175,177,179,181,183,185,187,189,191,193,195,197,199,201,203,205,207,209,211,213,215,217,219,221,223
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] flashinfer==0.1.6+cu121torch2.4
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-ml-py==12.560.30
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] pyzmq==26.2.0
[pip3] torch==2.5.0
[pip3] torchvision==0.20.0
[pip3] transformers==4.46.1
[pip3] triton==3.1.0
[conda] Could not collect
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.1.dev3243+g52666ce.d20241104 (git sha: 52666ce.d20241104
vLLM Build Flags:
CUDA Archs: Not Set; ROCm: Disabled; Neuron: Disabled
GPU Topology:
Could not collect

NVIDIA_VISIBLE_DEVICES=GPU-8c14074c-9922-6592-0c14-eec5bb4463e5,GPU-2492c185-f06f-4e97-4063-72067f27ee7b,GPU-882e27e1-4752-f6a0-bfc0-3e55ce274b8a,GPU-b49f0503-f439-921d-8124-e6029f30b83b
NVIDIA_REQUIRE_CUDA=cuda>=12.4 brand=tesla,driver>=470,driver<471 brand=unknown,driver>=470,driver<471 brand=nvidia,driver>=470,driver<471 brand=nvidiartx,driver>=470,driver<471 brand=geforce,driver>=470,driver<471 brand=geforcertx,driver>=470,driver<471 brand=quadro,driver>=470,driver<471 brand=quadrortx,driver>=470,driver<471 brand=titan,driver>=470,driver<471 brand=titanrtx,driver>=470,driver<471 brand=tesla,driver>=525,driver<526 brand=unknown,driver>=525,driver<526 brand=nvidia,driver>=525,driver<526 brand=nvidiartx,driver>=525,driver<526 brand=geforce,driver>=525,driver<526 brand=geforcertx,driver>=525,driver<526 brand=quadro,driver>=525,driver<526 brand=quadrortx,driver>=525,driver<526 brand=titan,driver>=525,driver<526 brand=titanrtx,driver>=525,driver<526 brand=tesla,driver>=535,driver<536 brand=unknown,driver>=535,driver<536 brand=nvidia,driver>=535,driver<536 brand=nvidiartx,driver>=535,driver<536 brand=geforce,driver>=535,driver<536 brand=geforcertx,driver>=535,driver<536 brand=quadro,driver>=535,driver<536 brand=quadrortx,driver>=535,driver<536 brand=titan,driver>=535,driver<536 brand=titanrtx,driver>=535,driver<536
NVIDIA_DRIVER_CAPABILITIES=compute,utility
VLLM_USAGE_SOURCE=production-docker-image
CUDA_VERSION=12.4.1
LD_LIBRARY_PATH=/usr/local/lib/python3.12/dist-packages/cv2/../../lib64:/usr/local/nvidia/lib:/usr/local/nvidia/lib64

```

</details>


### Model Input Dumps

_No response_

### 🐛 Describe the bug

When running 90b LLama 3.2 vision model using vllm, see command below:
python3 -m vllm.entrypoints.openai.api_server --served-model-name=meta-llama/Llama-3.2-90B-Vision-Instruct --model=/data/001 --tensor-parallel-size=4 --max-num-seqs=96 --max-log-len=0 --load-format=safetensors --host=0.0.0.0 --port=80 --max-num-seqs=64 --gpu-memory-utilization=0.95 --enforce-eager --max-model-len=32768

It occasionally crashes, see logs

INFO 11-25 16:02:04 logger.py:37] Received request cmpl-6ccba6262268459ab8a7317db4e6a4dd-0: prompt: '', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.7, top_p=0.9, top_k=10, min_p=0.0, seed=3497136763981421277, stop=['<|eot_id|>', '<|end_of_text|>', '<|eom_id|>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=1, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: [], lora_request: None, prompt_adapter_request: None.
INFO 11-25 16:02:04 logger.py:37] Received request chatcmpl-0f4e61f625404415ae19e6c65a805c48: prompt: '', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.3, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=['<|eot_id|>', '<|end_of_text|>', '<|eom_id|>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=750, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: None, lora_request: None, prompt_adapter_request: None.
INFO 11-25 16:02:05 engine.py:275] Added request cmpl-6ccba6262268459ab8a7317db4e6a4dd-0.
INFO 11-25 16:02:05 engine.py:275] Added request chatcmpl-0f4e61f625404415ae19e6c65a805c48.
INFO 11-25 16:02:05 logger.py:37] Received request chatcmpl-8599dad45f1d4e808952b46daef59e09: prompt: '', params: SamplingParams(n=1, presence_penalty=0.0, frequency_penalty=0.0, repetition_penalty=1.0, temperature=0.3, top_p=1.0, top_k=-1, min_p=0.0, seed=None, stop=['<|eot_id|>', '<|end_of_text|>', '<|eom_id|>'], stop_token_ids=[], bad_words=[], include_stop_str_in_output=False, ignore_eos=False, max_tokens=750, min_tokens=0, logprobs=None, prompt_logprobs=None, skip_special_tokens=True, spaces_between_special_tokens=True, truncate_prompt_tokens=None, guided_decoding=None), prompt_token_ids: None, lora_request: None, prompt_adapter_request: None.
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.54.175:43202 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.54.169:53022 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.51.218:57838 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.53.240:41778 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.10.36:41008 - "POST /v1/completions HTTP/1.1" 500 Internal Server Error
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.54.171:36942 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
CRITICAL 11-25 16:02:05 launcher.py:99] MQLLMEngine is already dead, terminating server process
INFO:     10.244.51.220:60982 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
ERROR 11-25 16:02:05 engine.py:143] RuntimeError('CUDA error: invalid configuration argument\nCUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.\nFor debugging consider passing CUDA_LAUNCH_BLOCKING=1\nCompile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.\n')
ERROR 11-25 16:02:05 engine.py:143] Traceback (most recent call last):
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 141, in start
ERROR 11-25 16:02:05 engine.py:143]     self.run_engine_loop()
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 204, in run_engine_loop
ERROR 11-25 16:02:05 engine.py:143]     request_outputs = self.engine_step()
ERROR 11-25 16:02:05 engine.py:143]                       ^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 222, in engine_step
ERROR 11-25 16:02:05 engine.py:143]     raise e
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/multiprocessing/engine.py", line 213, in engine_step
ERROR 11-25 16:02:05 engine.py:143]     return self.engine.step()
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/engine/llm_engine.py", line 1456, in step
ERROR 11-25 16:02:05 engine.py:143]     outputs = self.model_executor.execute_model(
ERROR 11-25 16:02:05 engine.py:143]               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/distributed_gpu_executor.py", line 82, in execute_model
ERROR 11-25 16:02:05 engine.py:143]     driver_outputs = self._driver_execute_model(execute_model_req)
ERROR 11-25 16:02:05 engine.py:143]                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/multiproc_gpu_executor.py", line 155, in _driver_execute_model
ERROR 11-25 16:02:05 engine.py:143]     return self.driver_worker.execute_model(execute_model_req)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/worker_base.py", line 343, in execute_model
ERROR 11-25 16:02:05 engine.py:143]     output = self.model_runner.execute_model(
ERROR 11-25 16:02:05 engine.py:143]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
ERROR 11-25 16:02:05 engine.py:143]     return func(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/enc_dec_model_runner.py", line 201, in execute_model
ERROR 11-25 16:02:05 engine.py:143]     hidden_or_intermediate_states = model_executable(
ERROR 11-25 16:02:05 engine.py:143]                                     ^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
ERROR 11-25 16:02:05 engine.py:143]     return self._call_impl(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1747, in _call_impl
ERROR 11-25 16:02:05 engine.py:143]     return forward_call(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/mllama.py", line 1363, in forward
ERROR 11-25 16:02:05 engine.py:143]     outputs = self.language_model(
ERROR 11-25 16:02:05 engine.py:143]               ^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
ERROR 11-25 16:02:05 engine.py:143]     return self._call_impl(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1747, in _call_impl
ERROR 11-25 16:02:05 engine.py:143]     return forward_call(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/mllama.py", line 1030, in forward
ERROR 11-25 16:02:05 engine.py:143]     hidden_states = self.model(
ERROR 11-25 16:02:05 engine.py:143]                     ^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
ERROR 11-25 16:02:05 engine.py:143]     return self._call_impl(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1747, in _call_impl
ERROR 11-25 16:02:05 engine.py:143]     return forward_call(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/mllama.py", line 962, in forward
ERROR 11-25 16:02:05 engine.py:143]     hidden_states = decoder_layer(
ERROR 11-25 16:02:05 engine.py:143]                     ^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
ERROR 11-25 16:02:05 engine.py:143]     return self._call_impl(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1747, in _call_impl
ERROR 11-25 16:02:05 engine.py:143]     return forward_call(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/mllama.py", line 880, in forward
ERROR 11-25 16:02:05 engine.py:143]     hidden_states = self.cross_attn(
ERROR 11-25 16:02:05 engine.py:143]                     ^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
ERROR 11-25 16:02:05 engine.py:143]     return self._call_impl(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1747, in _call_impl
ERROR 11-25 16:02:05 engine.py:143]     return forward_call(*args, **kwargs)
ERROR 11-25 16:02:05 engine.py:143]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/mllama.py", line 764, in forward
ERROR 11-25 16:02:05 engine.py:143]     output = self.attention_with_mask(q, k, v, kv_cache,
ERROR 11-25 16:02:05 engine.py:143]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143]   File "/usr/local/lib/python3.12/dist-packages/vllm/model_executor/models/mllama.py", line 807, in attention_with_mask
ERROR 11-25 16:02:05 engine.py:143]     self.head_dim).contiguous()
ERROR 11-25 16:02:05 engine.py:143]                    ^^^^^^^^^^^^
ERROR 11-25 16:02:05 engine.py:143] RuntimeError: CUDA error: invalid configuration argument
ERROR 11-25 16:02:05 engine.py:143] CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
ERROR 11-25 16:02:05 engine.py:143] For debugging consider passing CUDA_LAUNCH_BLOCKING=1
ERROR 11-25 16:02:05 engine.py:143] Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.
ERROR 11-25 16:02:05 engine.py:143] 
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229] Exception in worker VllmWorkerProcess while processing method start_worker_execution_loop.
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229] Traceback (most recent call last):
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/vllm/executor/multiproc_worker_utils.py", line 223, in _run_worker_process
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     output = executor(*args, **kwargs)
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]              ^^^^^^^^^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     return func(*args, **kwargs)
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]            ^^^^^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/worker_base.py", line 85, in start_worker_execution_loop
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     output = self.execute_model(execute_model_req=None)
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/worker_base.py", line 343, in execute_model
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     output = self.model_runner.execute_model(
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/torch/utils/_contextlib.py", line 116, in decorate_context
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     return func(*args, **kwargs)
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]            ^^^^^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/vllm/worker/enc_dec_model_runner.py", line 201, in execute_model
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     hidden_or_intermediate_states = model_executable(
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]                                     ^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1736, in _wrapped_call_impl
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     return self._call_impl(*args, **kwargs)
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]   File "/usr/local/lib/python3.12/dist-packages/torch/nn/modules/module.py", line 1747, in _call_impl
[1;36m(VllmWorkerProcess pid=352)[0;0m ERROR 11-25 16:02:05 multiproc_worker_utils.py:229]     return forward_call(*args, **kwargs)
...

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

@tjohnson31415 here is what I got from this log
```
diff --git a/vllm/model_executor/models/mllama.py b/vllm/model_executor/models/mllama.py
index 45f5dea08..248d95110 100644
--- a/vllm/model_executor/models/mllama.py
+++ b/vllm/model_executor/models/mllama.py
@@ -1407,7 +1407,12 @@ class MllamaForConditionalGeneration(nn.Module, SupportsMultiModal,
             ]
             for actual_len, last_group_len in zip(
                     actual_encoder_seq_lens, attn_metadata.encoder_seq_lens):
-                assert actual_len >= last_group_len
+                assert actual_len >= last_group_len, \
+                    (f"actual_len: {actual_encoder_seq_lens}, "
+                     f"last_group_len: {attn_metadata.encoder_seq_lens}, "
+                     f"num_tiles: {num_tiles}, "
+                     f"num_tokens_per_tile: {num_tokens_per_tile}, "
+                     f"image_size: {self.image_size}")
 
             cross_attention_states = self.get_cross_attention_states(
                 image_inputs, attn_metadata, actual_encoder_seq_lens)
```

There is a zero that shifts the array and triggers an assertion:
`(VllmWorkerProcess pid=475) ERROR 03-13 20:34:36 [multiproc_worker_utils.py:242] AssertionError: actual_len: [6404, 3202, 3202, 3202, 1601], last_group_len: [0, 6404, 3202, 3202, 3202, 1601], num_tiles: [[4], [2], [2], [2], [1]], num_tokens_per_tile: 1601, image_size: 560
`

--- a/tests/models/encoder_decoder/vision_language/test_mllama.py
+++ b/tests/models/encoder_decoder/vision_language/test_mllama.py
@@ -209,14 +209,15 @@ def _run_test(
     # will hurt multiprocessing backend with fork method (the default method).
 
     # max_model_len should be greater than image_feature_size
-    with vllm_runner(model,
-                     dtype=dtype,
-                     max_model_len=8192,
-                     max_num_seqs=3,
-                     tensor_parallel_size=tensor_parallel_size,
-                     distributed_executor_backend=distributed_executor_backend,
-                     limit_mm_per_prompt={"image": _LIMIT_IMAGE_PER_PROMPT
-                                          }) as vllm_model:
+    with vllm_runner(
+            model,
+            dtype=dtype,
+            max_model_len=19212,  # 3 max size images
+            max_num_seqs=3,
+            tensor_parallel_size=tensor_parallel_size,
+            distributed_executor_backend=distributed_executor_backend,
+            limit_mm_per_prompt={"image":
+                                 _LIMIT_IMAGE_PER_PROMPT}) as vllm_model:
         vllm_outputs_per_image = [
             vllm_model.generate_greedy_logprobs(prompts,
                                                 max_tokens,

--- a/tests/models/encoder_decoder/vision_language/test_mllama.py
+++ b/tests/models/encoder_decoder/vision_language/test_mllama.py
@@ -691,3 +691,26 @@ def test_get_full_text_row_masked_out_mask(input_indices) -> None:
                 f"full_text_row_masked_out_mask[{idx}] must be " \
                 f"'{must_be_masked}' "
             idx += 1
+
+
+@pytest.mark.core_model
+@pytest.mark.parametrize("encoder_seq_lens, num_tiles, expected", [
+    ([6404], [[4]], [6404]),
+    ([0, 6404], [[4]], [6404]),
+    ([0, 1601, 8005], [[1], [4, 1]], [1601, 8005]),
+    ([0, 19212, 0, 3202], [[4, 4, 4], [2]], [19212, 3202]),
+])
+def test_parse_and_validate_encoder_lens(encoder_seq_lens, num_tiles,
+                                         expected) -> None:
+
+    dummy = DummyModel()
+    num_tokens_per_tile = 1601
+    actual_encoder_seq_lens = MllamaForConditionalGeneration \
+        ._get_and_validate_encoder_lens(
+            dummy,
+            encoder_seq_lens,
+            num_tiles,
+            num_tokens_per_tile,
+        )
+    assert actual_encoder_seq_lens == expected, \
+        f"Expected {expected} but got {actual_encoder_seq_lens}"

--- a/tests/models/encoder_decoder/vision_language/test_mllama.py
+++ b/tests/models/encoder_decoder/vision_language/test_mllama.py
@@ -507,7 +507,7 @@ def test_regression(vllm_runner, image_assets, model, dtype, max_tokens,
             model,
             dtype=dtype,
             max_model_len=8192,
-            max_num_seqs=2,
+            max_num_seqs=4,
             tensor_parallel_size=1,
             limit_mm_per_prompt={"image":
                                  _LIMIT_IMAGE_PER_PROMPT}) as vllm_model:
@@ -546,6 +546,23 @@ def test_regression(vllm_runner, image_assets, model, dtype, max_tokens,
         images = [
             [stop_sign],
             None,
+        ]
+        vllm_model.generate_greedy_logprobs(prompts,
+                                            max_tokens,
+                                            num_logprobs,
+                                            images=images)
+
+        # Mixed batch with text and images with different numbers of tiles
+        prompts = [
+            "<|begin_of_text|>Hello!",
+            "<|begin_of_text|>Some text before.<|image|>What is in the image?",  # noqa: E501
+            "<|begin_of_text|>Some text before.<|image|>What is in the image?",  # noqa: E501
+        ]
+        images = [
+            None,
+            [stop_sign],
+            # smaller image must be 2nd for the repro
+            [stop_sign.resize((448, 448))],
         ]
         vllm_model.generate_greedy_logprobs(prompts,
                                             max_tokens,

--- a/vllm/model_executor/models/mllama.py
+++ b/vllm/model_executor/models/mllama.py
@@ -1447,8 +1447,14 @@ class MllamaForConditionalGeneration(nn.Module, SupportsMultiModal,
             actual_encoder_seq_lens = [
                 sum(num_tile) * num_tokens_per_tile for num_tile in num_tiles
             ]
-            for actual_len, last_group_len in zip(
-                    actual_encoder_seq_lens, attn_metadata.encoder_seq_lens):
+
+            # remove 0 entries for text-only requests for these assertions
+            attn_metadata_lens = [
+                len for len in attn_metadata.encoder_seq_lens if len > 0
+            ]
+            assert len(actual_encoder_seq_lens) == len(attn_metadata_lens)
+            for actual_len, last_group_len in zip(actual_encoder_seq_lens,
+                                                  attn_metadata_lens):
                 assert actual_len >= last_group_len
 
             cross_attention_states = self.get_cross_attention_states(

--- a/vllm/model_executor/models/mllama.py
+++ b/vllm/model_executor/models/mllama.py
@@ -1319,14 +1319,14 @@ class MllamaForConditionalGeneration(nn.Module, SupportsMultiModal,
         # Because attn_metadata.encoder_seq_lens only counts the last
         # group of images for each sample, which is used to cheat the
         # block manager to allocate blocks for those images only.
-        # See input_processor_for_mllama() for more details.
+        # See MllamaMultiModalProcessor for more details.
         actual_encoder_seq_lens = [
             sum(num_tile) * num_tokens_per_tile for num_tile in num_tiles
         ]
 
         # remove 0 encoder len entries for text-only requests for these
         # assertions
-        attn_metadata_lens = [len for len in encoder_seq_lens if len > 0]
+        attn_metadata_lens = [x for x in encoder_seq_lens if x > 0]
         assert len(actual_encoder_seq_lens) == len(attn_metadata_lens)
         for actual_len, last_group_len in zip(actual_encoder_seq_lens,
                                               attn_metadata_lens):

--- a/vllm/model_executor/models/mllama.py
+++ b/vllm/model_executor/models/mllama.py
@@ -1309,6 +1309,31 @@ class MllamaForConditionalGeneration(nn.Module, SupportsMultiModal,
 
         raise AssertionError("This line should be unreachable.")
 
+    def _get_and_validate_encoder_lens(
+        self,
+        encoder_seq_lens: List[int],
+        num_tiles: List[List[int]],
+        num_tokens_per_tile: int,
+    ) -> List[int]:
+        # Get the actual number of encoder tokens for each sample.
+        # Because attn_metadata.encoder_seq_lens only counts the last
+        # group of images for each sample, which is used to cheat the
+        # block manager to allocate blocks for those images only.
+        # See input_processor_for_mllama() for more details.
+        actual_encoder_seq_lens = [
+            sum(num_tile) * num_tokens_per_tile for num_tile in num_tiles
+        ]
+
+        # remove 0 encoder len entries for text-only requests for these
+        # assertions
+        attn_metadata_lens = [len for len in encoder_seq_lens if len > 0]
+        assert len(actual_encoder_seq_lens) == len(attn_metadata_lens)
+        for actual_len, last_group_len in zip(actual_encoder_seq_lens,
+                                              attn_metadata_lens):
+            assert actual_len >= last_group_len
+
+        return actual_encoder_seq_lens
+
     def flat_encoder_result(self, cross_attention_states: torch.Tensor,
                             attn_metadata: AttentionMetadata,
                             actual_encoder_seq_lens: List[int]):
@@ -1436,26 +1461,14 @@ class MllamaForConditionalGeneration(nn.Module, SupportsMultiModal,
         else:
             skip_cross_attention = False
 
-            # Get the actual number of encoder tokens for each sample.
-            # Because attn_metadata.encoder_seq_lens only counts the last
-            # group of images for each sample, which is used to cheat the
-            # block manager to allocate blocks for those images only.
-            # See MllamaMultiModalProcessor for more details.
-            num_tiles_tensor = kwargs.pop("num_tiles")
-            num_tiles = [t.tolist() for t in num_tiles_tensor]
+            num_tiles = [t.tolist() for t in kwargs.pop("num_tiles")]
             num_tokens_per_tile = calc_token_per_chunk(self.image_size)
-            actual_encoder_seq_lens = [
-                sum(num_tile) * num_tokens_per_tile for num_tile in num_tiles
-            ]
 
-            # remove 0 entries for text-only requests for these assertions
-            attn_metadata_lens = [
-                len for len in attn_metadata.encoder_seq_lens if len > 0
-            ]
-            assert len(actual_encoder_seq_lens) == len(attn_metadata_lens)
-            for actual_len, last_group_len in zip(actual_encoder_seq_lens,
-                                                  attn_metadata_lens):
-                assert actual_len >= last_group_len
+            actual_encoder_seq_lens = self._get_and_validate_encoder_lens(
+                attn_metadata.encoder_seq_lens,
+                num_tiles,
+                num_tokens_per_tile,
+            )
 
             cross_attention_states = self.get_cross_attention_states(
                 image_inputs, attn_metadata, actual_encoder_seq_lens)

